runtime/lua/nvim/spellfile.lua

```diff
@@ -0,0 +1,278 @@
+    notify('Created ' .. target)
+  end
+  return target
+end
+
+local function file_ok(path)
+  local s = vim.uv.fs_stat(path)
+  return s and s.type == 'file' and (s.size or 0) > 0
+end
+
+local function reload_spell_silent()
+  vim.cmd('silent! setlocal spell!')
+  if vim.bo.spelllang and vim.bo.spelllang ~= '' then
+    vim.cmd('silent! setlocal spelllang=' .. vim.bo.spelllang)
+  end
+  vim.cmd('echo ""')
+end
+
+--- Blocking GET to file with timeout; treats status==0 as success if file exists.
+--- @return boolean ok, integer|nil status, string|nil err
+local function http_get_to_file_sync(url, outpath, timeout_ms)
+  local done, err, res = false, nil, nil
+  vim.net.request(url, { outpath = outpath }, function(e, r)
+    err, res, done = e, r, true
+  end)
+  vim.wait(timeout_ms or M.config.timeout_ms, function()
+    return done
+  end, 50, false)
+
+  local status = res and res.status or 0
+  local ok = (not err) and ((status >= 200 and status < 300) or (status == 0 and file_ok(outpath)))
+  return ok, (status ~= 0 and status or nil), err
+end
+
+---@return string[]
+function M.directory_choices()
+  local opts = {}
+  for _, dir in ipairs(rtp_list()) do
+    local spelldir = vim.fs.joinpath(vim.fn.fnamemodify(dir, ':p'), 'spell')
+    if vim.fn.isdirectory(spelldir) == 1 then
+      table.insert(opts, spelldir)
+    end
+  end
+  return opts
+end
+
+function M.choose_directory()
+  local dirs = writable_spell_dirs_from_rtp()
+  if #dirs == 0 then
+    return ensure_target_dir()
+  elseif #dirs == 1 then
+    return dirs[1]
+  end
+  local prompt ---@type string[]
+  prompt = {}
+  for i, d in
+    ipairs(dirs --[[@as string[] ]])
+  do
+    prompt[i] = string.format('%d: %s', i, d)
+  end
+  local choice = vim.fn.inputlist(prompt)
+  if choice < 1 or choice > #dirs then
+    return dirs[1]
+  end
+  return dirs[choice]
+end
+
+function M.parse(lang)
+  local code = normalize_lang(lang)
+  local enc = 'utf-8'
+  local dir = ensure_target_dir()
+
+  local missing = {}
+  local candidates = {
+    string.format('%s.%s.spl', code, enc),
+    string.format('%s.%s.sug', code, enc),
+  }
+  for _, fn in ipairs(candidates) do
+    if not file_ok(vim.fs.joinpath(dir, fn)) then
+      table.insert(missing, fn)
+    end
+  end
+
+  return {
+    files = missing,
+    key = code .. '.' .. enc,
+    lang = code,
+    encoding = enc,
+    dir = dir,
+  }
+end
+
+---@param info SpellInfo
+function M.download(info)
+  local dir = info.dir or ensure_target_dir()
+  if not dir then
+    notify('No (writable) spell directory found and could not create one.', vim.log.levels.ERROR)
+    return
+  end
+
+  local lang = info.lang
+  local enc = info.encoding
+
+  local spl_utf8 = string.format('%s.%s.spl', lang, enc)
+  local spl_ascii = string.format('%s.ascii.spl', lang)
+  local sug_name = string.format('%s.%s.sug', lang, enc)
+
+  local url_utf8 = M.config.url .. '/' .. spl_utf8
+  local out_utf8 = vim.fs.joinpath(dir, spl_utf8)
```

Comment:
when I started with `NVIM_APPNAME=foo nvim` and tried `set spelllang=fr` i see: 

```
No spell file found for fr (utf-8). Download? [y/N] yDownloading fr.utf-8.spl …
Saved fr.utf-8.spl to <runtimepath>/neovim/runtime/spell/fr.utf-8.spl
Downloading fr.utf-8.sug …
Saved fr.utf-8.sug to <runtimepath>/neovim/runtime/spell/fr.utf-8.sug
```

why didn't it create a `spell/` dir in the user-local stdpath()?